libraries/rush-lib/src/logic/ProjectWatcher.ts

```diff
@@ -88,20 +94,25 @@ export class ProjectWatcher {
     // https://nodejs.org/docs/latest-v12.x/api/fs.html#fs_caveats
     const useNativeRecursiveWatch: boolean = os.platform() === 'win32' || os.platform() === 'darwin';
 
-    for (const project of this._projectsToWatch) {
-      const projectState: Map<string, string> = (await previousState._tryGetProjectDependenciesAsync(
-        project,
-        this._terminal
-      ))!;
-      const projectFolder: string = project.projectRelativeFolder;
-      // Watch files in the root of the project, or
-      for (const fileName of projectState.keys()) {
-        for (const pathToWatch of ProjectWatcher._enumeratePathsToWatch(
-          fileName,
-          projectFolder,
-          useNativeRecursiveWatch
-        )) {
-          pathsToWatch.add(`${repoRoot}/${pathToWatch}`);
+    if (useNativeRecursiveWatch && this._projectsToWatch.size > 10) {
+      // Watch the entire repository; a single recursive watcher is cheap.
+      pathsToWatch.add(this._repoRoot);
+    } else {
```

Comment:
Is there a good reason to not always do this?

Reply:
I'm honestly not sure. My thought had been related to noise from unrelated changes kicking the detector more than needed, but I could just enable it all the time.